index.js

```diff
@@ -642,6 +691,27 @@
     return true;
   }
 
+  //  Array$prototype$lte :: Array a ~> Array a -> Boolean
+  function Array$prototype$lte(other) {
+    for (var idx = 0; idx < this.length; idx += 1) {
+      var a = this[idx], b = other[idx];
+
+      // I don't like the two checks here. If this is a nested structure,
+      // the call to equals could have been expensive. A call to lte will be
+      // just as expensive.
+      // With compare it would be:
+      // try {
+      //   var cmp = compare(this[idx], other[idx]);
+      //   if (cmp === 'LT' || cmp === 'EQ') return true;
+      //   if (cmp === 'GT') return false;
+      // } catch() { return false; }
+      if (!equals(a, b)) {
+        return a === void 0 ? true : b === void 0 ? false : lte(a, b);
```

Comment:
`xs[idx] === undefined` is not a reliable way to determine an array's bounds. It introduced a bug, in this case:

```javascript
> Z.lte([], [undefined])
true

> Z.lte([undefined], [])
true
```

Let's reinstate the `length` checks.